include/swift/AST/Attr.h

```diff
@@ -730,6 +731,63 @@ class SectionAttr : public DeclAttribute {
   }
 };
 
+/// Defines the @performanceOverride attribute.
+class PerformanceOverrideAttr : public DeclAttribute {
+public:
+  enum class CheckKind : uint8_t { ReturnTypeImplicitCopy, Count = 1 };
```

Comment:
I'm not really a fan of the design here that uses an attribute to silence warnings. IMO, either there's a general feature to silence warnings within a region of the source, or we don't do it.

Reply:
Removed the attribute from this PR to consider a more general design. I'd like to proceed with just the new diagnostic logic for now.